source/core_json.c

```diff
@@ -1283,33 +1348,210 @@ static JSONStatus_t search( char * buf,
         }
     }
 
-    if( ret == JSONSuccess )
+    if( ret == true )
     {
-        /* String values and collections include their surrounding
-         * demarcation.  If the value is a string, strip the quotes. */
-        if( buf[ value ] == '"' )
+        *outValue = &buf[ value ];
+        *outValueLength = valueLength;
+    }
+
+    return ret;
+}
+
+/**
+ * @brief Find an index in a JSON array and output a pointer to its value.
+ *
+ * @param[in] buf  The buffer to search.
+ * @param[in] max  size of the buffer.
+ * @param[in] queryIndex  The index to search for.
+ * @param[out] outValue  A pointer to receive the address of the value found.
+ * @param[out] outValueLength  A pointer to receive the length of the value found.
+ *
+ * Iterate over the values of an array, looking for a matching index.
+ *
+ * @return true if the queryIndex is found and the value output;
+ * false otherwise.
+ *
+ * @note Parsing stops upon finding a match.
+ */
+static bool_ arraySearch( char * buf,
+                          size_t max,
+                          uint32_t queryIndex,
+                          char ** outValue,
+                          size_t * outValueLength )
+{
+    bool_ ret = false;
+    size_t i = 0, value, valueLength;
+    uint32_t j = 0;
```

Comment:
`j` is not easy to follow...
```suggestion
    uint32_t arrayIndex = 0;
```

Reply:
Changed to `currentIndex`.